policies.go

```diff
@@ -311,6 +311,9 @@ type HostSelectionPolicy interface {
 	SetTablets
 	KeyspaceChanged(KeyspaceUpdateEvent)
 	Init(*Session)
+	// Reset is opprotunity to reset HostSelectionPolicy if Session initilization failed and we want to
+	// call HostSelectionPolicy.Init() again with new Session
+	Reset()
```

Comment:
As food for thought, maybe it is better to name it so that it is clear from name that it is going to be called only when `HostSelectionPolicy` ends it's life-cycle, something around `Stop` or `Close`